### PR TITLE
Emily robot state

### PR DESCRIPTION
**Pull request by emilycxliu**

### CODE REVIEW

**Comment by weizhang-2706** [comment]
A lot of previous path planner changes (which are already merged) in this pull request.
Need to synchronize with main branch first.

File: src/main/java/frc/robot/robotcontainers/CompRobotContainer.java [hunk 1]
```diff
@@ -20,6 +20,13 @@
  */
 public class CompRobotContainer extends RobotContainer {
 
+  public enum RobotState {
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/robotcontainers/CompRobotContainer.java:23]: WPIlib already has a class of RobotState for driver station's state.
Suggested to use a different name, e.g., RobotGamePieceState.

File: src/main/java/frc/robot/commands/SelectByRobotState.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+package frc.robot.commands;
+
+import java.util.function.Supplier;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.robotcontainers.CompRobotContainer.RobotState;
+
+public class SelectByRobotState extends CommandBase {
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/SelectByRobotState.java:13]: GamePiece state can be used by the arm command directly.  We don't need a separate command.

**Review by weizhang-2706 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/SelectByRobotState.java (hunk 2 above)
Comment (ErikCald) [reply src/main/java/frc/robot/commands/SelectByRobotState.java:13]: I suppose we could integrate GamePieceState directly into an ArmCommand

Comment (weizhang-2706) [reply src/main/java/frc/robot/commands/SelectByRobotState.java:13]: Yes, ArmCommand will have a supplier arg of GamePieceState and GripperCommand(intake) will have a consumer arg of GamePieceSate.

**Pull request closed without merging**